src/main/java/org/spongepowered/api/item/inventory/BlockCarrier.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.util.Direction;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Represents a Block that is a carrier of an Inventory
+ * e.g. An Anvil
+ */
+public interface BlockCarrier extends Carrier {
+
+    /**
+     * Returns the Locations of the Blocks
+     *
+     * @return The Locations of the Blocks
+     */
+    List<Location<World>> getLocations();
```

Comment:
I'd prefer the original way; it makes it harder for plugin developers to write (forgive the wording) dumb code. With the multiple block type as a subtype, it's very easy for plugins to forget there are multiple blocks; the correct code to get all blocks would be this:

```java
Collection<Location> locations;
if (carrier instanceof MultiBlockCarrier) {
    locations = ((MultiBlockCarrier) carrier).getLocations();
} else {
    locations = Collection.singleton(carrier.getLocation();
}
```
On the other hand, it's easy to forget to use all this - using simply `carrier.getLocation()` could result in incorrect behavior.

On the other hand, in the original method, plugin developers would have to either handle multiple locations or cast to force it to only be one location. It's still possible to try working around it, but it's an extra step to nudge developers in the right direction.